glean-core/android/src/main/java/mozilla/telemetry/glean/private/StringMetricType.kt

```diff
@@ -73,6 +73,15 @@ class StringMetricType internal constructor(
         LibGleanFFI.INSTANCE.glean_string_set(this.handle, value)
     }
 
+    /**
+     * Internal only. Check whether a field has a value. Used for determining if
+     * all core metrics have been set.
+     */
+    internal fun hasValue(): Boolean {
```

Comment:
Brrrrrr please let's not use this. If crashes are really the culprit here, we can also just rely on the dirty flag.

Reply:
Unfortunately, the dirty flag isn't sufficient here.  We are still "ok" with these pings if the app follows:

"foreground (dirty) -> background -> successfully sends ping (not dirty) -> app shutdown"

I thought about adding another flag, but thought that would be even worse.

In any event, I think I'm preferring the approach in #1123 at this point anyway...